### PR TITLE
docs(site): Rebuild documentation site

### PR DESCRIPTION
**Pull request by markdalgleish**
It's taken a lot of work, but the PR is finally here! 🎉 

The current documentation site is still largely a hold-over from our hackathon project. This has resulted in a few critical issues that this PR is designed to address:
- The site isn't responsive, and the navigation has basically run out of room.
- Adding new documentation requires a *lot* of code, generally created with a *lot* of manual copying-and-pasting, making review and maintenance extremely difficult.

To solve the first point, the site has been significantly redesigned with mobile in mind.

The second point has been addressed by making component demos data-driven rather than being a series of pages. To achieve this, I've introduced the concept of a "demo spec" which sits alongside the component, rather than being inside the `docs` folder. This greatly reduces the overhead creating and maintaining component demos, and allows us to easily redesign component demos across the board.

There's still a lot more work to go on this, but this PR is designed to give us a much more solid foundation to build from.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -62,7 +63,7 @@
     "no-floating-decimal": 2,
     "no-implicit-coercion": 2,
     "no-implied-eval": 2,
-    "no-invalid-this": 2,
+    "no-invalid-this": 0,
```
Comment (michaeltaranto) [inline .eslintrc]: How does this rule interplay with the `babel/no-invalid-this` rule?

Comment (markdalgleish) [inline .eslintrc]: The babel rule is meant to be a replacement for this one.

File: webpack.static.render.config.js [hunk 2]
```diff
@@ -12,14 +14,14 @@ const template = ejs.compile(fs.readFileSync(templatePath, 'utf-8')); // eslint-
...
-  path.resolve(__dirname, 'app'),
+  path.resolve(__dirname, BASE_DIR),
   path.resolve(__dirname, 'wip_modules')
 ];
 
 const routes = [
   '/',
   '/playground',
-  '/icons',
+  // '/icons',
```
Comment (mengtzu) [inline webpack.static.render.config.js]: Maybe just delete?

File: docs/components/Home/Hero/ColorPreview/Droplet/Droplet.less [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  @media only screen and (min-width: 500px) {
+    display: none;
+  }
+  @media only screen and (min-width: 580px) {
+    display: block;
+  }
+  @media only screen and (min-width: 640px) {
+    display: none;
+  }
+  @media only screen and (min-width: 740px) {
```
Comment (mengtzu) [inline docs/components/Home/Hero/ColorPreview/Droplet/Droplet.less]: At least one of them is standard :trollface:

**Comment by mengtzu** [comment]
🎨  📚  :burgerparrot:

**Pull request merged**